nikola/__main__.py

```diff
@@ -122,9 +121,24 @@ def main(args=None):
 
     sys.path.append('')
     try:
-        loader = importlib.machinery.SourceFileLoader("conf", conf_filename)
-        conf = loader.load_module()
-        config = conf.__dict__
+        @contextmanager
```

Comment:
Why is this an inline function?

Reply:
Well honestly I was not that sure whether or not to make this function local.
I decided to make it local because this is the only spot where this function is used.

Should I move it to the global scope?

Reply:
Could there be a way to avoid doing it via a contextmanager, and instead using in-place `try`-`finally`?

Reply:
The alternative to `@contextmanager` would be to create a class with magic methods called `__enter__` and `__exit__`.
This would look sort of like this:

```py
class TempImporter(filename):
    def __enter__(self):
        self.old_path = sys.path
        self.old_modules = sys.modules
        sys.path = sys.path[:]
        sys.modules = sys.modules.copy()
        sys.path.insert(0, os.path.dirname(filename))
        with open(filename, "rb") as file:
            return imp.load_module(filename, file, filename, (None, "rb", imp.PY_SOURCE))

    def __exit__(self, exc_type, exc_val, exc_tb):
        sys.path = self.old_path
        sys.modules = self.old_modules
```
and
```py
with TempImporter(conf_filename) as module:
    config = module.__dict__
```

Which doesn't look that smooth to me.
What would you recommend me to do?

Reply:
[8 PEP 343: The 'with' statement](https://docs.python.org/2.5/whatsnew/pep-343.html)

Reply:
Is there any problem with removing the `with` statement altogether?

```py
    sys.path.append('')
    try:
        old_path = sys.path
        old_modules = sys.modules
        sys.path = sys.path[:]
        sys.modules = sys.modules.copy()
        sys.path.insert(0, os.path.dirname(filename))
        try:
            with open(filename, "rb") as file:
                module = imp.load_module(filename, file, filename, (None, "rb", imp.PY_SOURCE))
            config = module.__dict__
        finally:
            sys.path = old_path
            sys.modules = old_modules
    except Exception:
```

Reply:
Ooh yeah you're right!
I've got a little confused by solutions on stack-overflow which all were using the `with` statement.

I'll just go on and remove them